src/Messages/Interactions/TSInteraction.h

```diff
@@ -33,7 +32,18 @@ NS_ASSUME_NONNULL_BEGIN
 + (instancetype)interactionForTimestamp:(uint64_t)timestamp
                         withTransaction:(YapDatabaseReadWriteTransaction *)transaction;
 
-- (nullable NSDate *)receiptDateForSorting;
+// NSDate has second precision, uint64_t timestamps have millisecond precision
+// so prefer timestampForSorting over dateForSorting.
+- (NSDate *)dateForSorting;
```

Comment:
I'm OK with this, and further, it's clearly easy to forget about precision when dealing in seconds units, given the number of bugs you fixed around this. I think dealing with the raw milliseconds makes this a little more me-proof.